feature_engine/selection/drop_correlated_features.py

```diff
@@ -73,6 +77,15 @@ class DropCorrelatedFeatures(BaseSelector):
         The correlation threshold above which a feature will be deemed correlated with
         another one and removed from the dataset.
 
+    order_by: str, default=None
+        Type of sorting to use before feature selection. This option could help with
+        the consistency of the selection.
+
+        - None - preserve original order of the dataframe
+        - 'nan' - sort columns by number of missing values (ascending)
+        - 'unique' - sort columns by number of unique values (descending)
+        - 'alphabetic' - sort columns alphabetically.
```

Comment:
I am thinking the following:

- alphabetic isn't really a meaningful way of ordering. It has nothing to do with the variable's predictive value. It helps for consistency and is easy and fast, but other than that it does not add much
- nan has some value in that it will preserve the variables with less nan, but it would be useful only before imputation and when data does have a lot of nan.
- unique will take ages if all variables are continuous

Hence, I think it might be useful to add the additional option to use the std, and do tell users that the limitation is the feature scale, so to scale features when using this option.

Would you agree?

Reply:
Std would be bad if:
 - you have features with different scales;
 - you have dtypes that pandas does not count as numerical and will ignore them;
In first case it won't be useful for the transformer, in second case it will fail with not very informative message. While sort by stdev sounds good, after tests I'm not a big fan of it.

All of this options would have drawbacks, the idea is to be consistent and transparent about in what order features are passed through elimination process.

Reply:
For 1mil rows and 8 features:

`297 ms ± 3.33 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)`

Reply:
Thank you. 

Yes, I got up thinking that also if you use the standard scaler, then the std should be close to 1 for all vars so dies not make a lot of sense either.